File: algorithms/python3/219_contains_duplicate_II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hash=dict()

        for i in range(len(nums)):
            if nums[i] in hash:
                if i-hash[nums[i]]<=k:
                    return True
            hash[nums[i]]=i
        return False

a=Solution()

print(a.containsNearbyDuplicate([1,2,3,1],3))

File: algorithms/cpp/219_contains_duplicate_II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        if (nums.empty())
            return false;
        unordered_map<int,int> index_map;
        for (auto it=nums.begin(); it != nums.end();it++){
            auto map_it=index_map.find(*it);
            if (map_it != index_map.end() && it-nums.begin()-map_it->second<=k)
            return true;
            index_map[*it]=it-nums.begin();
        }
        return false;
    }
};
